src/nodes/materials/NodeMaterial.js

```diff
@@ -351,6 +351,20 @@ class NodeMaterial extends Material {
 
 	}
 
+	setupLightMap( builder ) {
+
+		let node = null;
+
+		if ( builder.material.lightMap ) {
+
+			node = new IrradianceNode( materialLightMap );
```

Comment:
Maybe we sould move this to StandardMeshNodeMaterial like `setupEnvironment()`?

Reply:
I've asked this myself this as well but then I've realized when I do this `setupLightMap()` in `NodeMaterial` will be empty. The basic and standard materials share no common code in context of light maps like the setup of the environment map node in `NodeMaterial.setupEnvironment()`. Besides, we would have to implement `setupLightMap()` in phong, lambert and toon not just in `MeshStandardNodeMaterial`.

The light map handling in `MeshBasicNodeMaterial` is quite an exception and I thought it's more clear to completely overwrite the standard implementation from `NodeMaterial` . On the other side I'm not feeling strong about this issue. If you think it's better, I'll create a `setupLightMap()` method for each material that supports light maps.